### PR TITLE
Add support for merge commits

### PR DESCRIPTION
**Pull request by janos-ss**

### CODE REVIEW

File: sync_public_master.sh [hunk 1]
```diff
@@ -132,6 +143,8 @@ info "Cherry-picking from master_work into public_master_work..."
 pause
 git cherry-pick --keep-redundant-commits --allow-empty --strategy=recursive -X ours ${LATEST_MASTER_REF}..master_work
 
+validate_public_equivalent_refs "public_master" "master"
```
Comment (sns-seb) [inline sync_public_master.sh]: why do it before the last `filter-branch`?

This `filter-branch` could have unexpected side effect that we won't be able to catch checking the consistency too early

File: sync_public_master.sh [hunk 2]
```diff
@@ -128,25 +165,28 @@ git filter-branch -f --prune-empty --index-filter 'git rm --cached --ignore-unma
...
+validate_public_equivalent_refs "HEAD" "master"
 
-# log created references
-git for-each-ref --count=2 --sort=-refname "${REF_TREE_ROOT}"
+if [ ! "$cherry_pick_failed" ]; then
+  rm "$cherry_pick_sh"
+fi
 
-info "done"
+"$script_dir/finish_sync_public_master.sh"
```
Comment (sns-seb) [inline sync_public_master.sh]: I don't see any such file in the repository

File: sync_public_master.sh [hunk 3]
```diff
@@ -17,6 +17,28 @@
 
 set -euo pipefail
 
+script_dir=$(dirname "${BASH_SOURCE[0]}")
+
+cherry_pick_failed=
+cherry_pick_sh=cherry-pick.$$.sh
+
+cat << "EOF" > "$cherry_pick_sh"
```
Comment (sns-seb) [inline sync_public_master.sh]: what this does is pretty obscure to me
a few comment will help me now and anyone in the future

File: sync_public_master.sh (hunk 1 above)
Comment (janos-ss) [reply sync_public_master.sh]: Indeed, better to do later. (Forgot to address this earlier, it didn't help that GitHub collapsed these comments together with the other.)

**Pull request merged**